### ISSUE TITLE
[BUG] - Conda-store environments still reporting "building" after 24 hours

### ISSUE THREAD

**Issue by rsignell-usgs** [nebari-dev/nebari#1559]
### Describe the bug

I have several environments that still say "building" after 24 hours:
![2022-11-18_16-55-46](https://user-images.githubusercontent.com/1872600/202812683-71712855-42bc-4b16-8738-5d3be26adfe6.png)

If I ssh into the `conda-store-worker` pod, I see this when I run `top`:
![2022-11-18_17-00-08](https://user-images.githubusercontent.com/1872600/202812751-7d11da80-9f58-4ef4-b363-0f2cbc12169b.png)

I'm not sure if this is a bug.    How best to proceed?

### Expected behavior

environments build successfully or fail 

### OS and architecture in which you are running Nebari

Linux, AWS

### How to Reproduce the problem?

Not sure

### Versions and dependencies used.

Nebari (qhub) 0.4.4

### Compute environment

AWS

### Integrations

Keycloak, conda-store, Dask, Argo

**Comment by iameskild** [comment]
Hi @rsignell-usgs, thanks for bringing this to our attention! I've not seen this issue before so I'm unsure what the underlying cause might be. Can you please provide logs for the `conda-store-worker` and `conda-store-server` pods? Do you also happen to know how much space the conda-store storage has remaining?

cc @costrouc

**Comment by rsignell-usgs** [comment]
@iameskild thanks for taking a look at this. 

conda-store is only 25% full (I've been regularly keeping a watch on this and deleting old environments)

conda-store-worker log
``` 
 conda-store-worker [2022-11-21 14:10:00,746: ERROR/ForkPoolWorker-4] Task task_build_conda_docker[4f2b0176-2e99-4c69 ││ conda-store-worker Traceback (most recent call last):                                                                ││ conda-store-worker   File "/opt/conda/envs/conda-store-server/lib/python3.10/site-packages/celery/app/trace.py", lin ││ conda-store-worker     R = retval = fun(*args, **kwargs)                                                             ││ conda-store-worker   File "/opt/conda/envs/conda-store-server/lib/python3.10/site-packages/celery/app/trace.py", lin ││ conda-store-worker     return self.run(*args, **kwargs)                                                              ││ conda-store-worker   File "/opt/conda-store-server/conda_store_server/worker/tasks.py", line 114, in task_build_cond ││ conda-store-worker     build_conda_docker(conda_store, build)                                                        ││ conda-store-worker   File "/opt/conda-store-server/conda_store_server/build.py", line 296, in build_conda_docker     ││ conda-store-worker     image = build_docker_environment_image(                                                       ││ conda-store-worker   File "/opt/conda/envs/conda-store-server/lib/python3.10/site-packages/conda_docker/conda.py", l ││ conda-store-worker     add_conda_layers(                                                                             ││ conda-store-worker   File "/opt/conda/envs/conda-store-server/lib/python3.10/site-packages/conda_docker/conda.py", l ││ conda-store-worker     add_conda_package_layers(                                                                     ││ conda-store-worker   File "/opt/conda/envs/conda-store-server/lib/python3.10/site-packages/conda_docker/conda.py", l ││ conda-store-worker     with open(meta_path) as f:                                                                    ││ conda-store-worker FileNotFoundError: [Errno 2] No such file or directory: '/tmp/tmp8ixguert/opt/conda/conda-meta/_l ││                                        
```
conda-store-server pod logs seem fine.

**Comment by aktech** [comment]
@rsignell-usgs can you also share the environment, just in case if we can reproduce from that.

**Comment by pavithraes** [comment]
@rsignell-usgs Thanks again for reporting this! As `@aktech` mentioned, it'll be helpful to be able to reproduce it.

Since it has been a while, could you please share your environment file if you see this happening again?

We can leave this open for a couple of weeks, and then close it if nobody sees this behavior. We can always re-open if needed. :)

**Comment by rsignell-usgs** [comment]
Sorry, this one slipped off my radar and we ended up destroying this deploy due to other issues.

**Comment by rsignell** [comment]
I'm hitting this problem again with my Open Science Computing Nebari deployment (2024.3.2).  
I had successfully built a large environment in conda-store in about 17 minutes.  I then added one additional package to install via pip and it's been "building" for the last 24 hours (still building).  

How best to handle this?

**Comment by rsignell** [comment]
I clicked "d" on the pod above shown in red, and it says:
```
  Name:             nebari-conda-store-worker-788d56769-6bwcm                                                                                                            ││ Namespace:        dev                                                                                                                                                  ││ Priority:         0                                                                                                                                                    ││ Service Account:  default                                                                                                                                              ││ Node:             ip-10-10-12-87.us-west-2.compute.internal/10.10.12.87                                                                                                ││ Start Time:       Tue, 02 Apr 2024 12:28:00 -0400                                                                                                                      ││ Labels:           pod-template-hash=788d56769                                                                                                                          ││                   role=nebari-conda-store-worker                                                                                                                       ││ Annotations:      checksum/conda-environments: 2d15f1442d9df32f8b9d4a0014138bcf4265525c6c73d736b512184172ee577a                                                        ││                   checksum/config-map: f42edd8842a29a053d3cff0699a39ab86ba84dae9ffe2d5bddb2538c8076a914                                                                ││                   checksum/secret: d4a04294823ea201c0a41f0abc0a3a4454b37fad5906f66a2b8b5f52e51973fc                                                                    ││ Status:           Failed                                                                                                                                               ││ Reason:           Evicted                                                                                                                                              ││ Message:          The node was low on resource: ephemeral-storage. Threshold quantity: 5367448860, available: 5096308Ki. Container nfs-server was using 24Ki, request  ││ is 0, has larger consumption of ephemeral-storage. Container conda-store-worker was using 33310300Ki, request is 0, has larger consumption of ephemeral-storage.       ││ IP:               10.10.16.60                                                                                                                                          ││ IPs:                                                                                                                                                                   ││   IP:           10.10.16.60                                                                                                                                            ││ Controlled By:  ReplicaSet/nebari-conda-store-worker-788d56769                                                                                                         ││ Containers:                                                                                                                                                            ││   conda-store-worker:                                                                                                                                                  ││     Container ID:                                                                                                                                                      ││     Image:         quansight/conda-store-server:2024.3.1                                                                                                               ││     Image ID:                                                                                                                                                          ││     Port:          <none>                                                                                                                                              ││     Host Port:     <none>                                                                                                                                              ││     Args:                                                                                                                                                              ││       conda-store-worker                                                                                                                                               ││       --config                                                                                                                                                         ││       /etc/conda-store/conda_store_config.py                                                                                                                           ││     State:          Terminated                                                                                                                                         ││       Reason:       ContainerStatusUnknown                                                                                                                             ││       Message:      The container could not be located when the pod was terminated                                                                                     ││       Exit Code:    137                                                                                                                                                ││       Started:      Mon, 01 Jan 0001 00:00:00 +0000                                                                                                                    ││       Finished:     Mon, 01 Jan 0001 00:00:00 +0000                                                                                                                    ││     Last State:     Terminated                                                                                                                                         ││       Reason:       ContainerStatusUnknown                                                                                                                             ││       Message:      The container could not be located when the pod was deleted.  The container used to be Running                                                     ││       Exit Code:    137                                                                             
```

**Comment by rsignell** [comment]
After deleting the pod shown in red, things seemed to be okay with k9s (no red pods).   
I then tried building my environment again, and it said "building".   But then I noticed that the previous build also still said "building".    And then neither one finished building (I waited a few hours).     

 I tried logging in 24 hours later to see if anything had finished or changed, I found I could not even login to conda store, and k9s looks like this:
![image](https://github.com/nebari-dev/nebari/assets/125569335/234222a2-dbc9-461f-8537-97e4bbbf2281)

**Comment by viniciusdc** [comment]
Oh, that's something you don't see everyday.

There are two main things going on right now:
- The conda-store issue you are facing of stuck builds and the access to the webpage
- The services with red name

For the conda-store:
- I would get the logs from the worker and check if there are any errors there. I assume you might see something regarding an SQL inclusion error

actions:
- restart the conda-store-server pod (wait for the service to be back around 30s)
- go into /conda-store, copy the env yaml for the current stuck envs to some other place in your computer
delete those environments (unless you want the version history of those)
- restart conda-store-worker pod
- submit the new builds for the environments using the create env workflow

Now, the red services. Usually the main causes for this are:
- Incorrect mounting or volume errors (I don't think its the case here)
- Issues when running the health checks on the exported ports

When this happens, the important thing is to make sure it's in a running state, which is the case for almost all of the services there. Good thing.

I am concerned by the ebs-csi-controller though as its usually associated to scaling and node stuff on aws/k8s

**Comment by rsignell** [comment]
@viniciusdc I tried to get the logs from k9s, but every "red" pod I tried refused to return logs.   I didn't see anything either doing "d" on the pods.     

From the console, I did notice that the dbs-csi-controller referenced something about kubernetes 1.39, while my nebari 2024.03.02 config is still specifying 1.26.      

So my plan is to:
* destroy the 2024.03.02 deployment
* upgrade the config to 2024.03.03
* make sure the upgrade process updates kubernetes in the config
* deploy 2024.03.03
* try to create the environment again

**Comment by rsignell** [comment]
I destroyed the 2024.03.02 deployment, upgraded the config, changed kubernetes from 1.26 to 1.29 in the config (the upgrade process apparently didn't do that), and deployed 2024.03.03.

The deployment went fine, but when I tried to create the environment again, the build in conda-store stayed pending for 4 hours (when I logged into the conda-store worker, the cpu was 100% and memory use 5%).

So I destroyed again.   

Here is the [environment.yml](https://gist.github.com/rsignell/792e4fc0d5f594b166e4582604356945) that I'm trying to build. 

The crazy thing is it built okay without the last package (the `cmip6-downscaling` package)
